lib/generator.dart

```diff
@@ -342,6 +342,11 @@ Make sure your query is correct and your schema is updated.''');
     annotations.add('JsonKey(name: \'${name.name}\')');
   }
 
+  final fieldDirectives =
+      regularField?.directives ?? regularInputField?.directives;
+
+  _addDeprecatedAnnotationIfNecessary(annotations, fieldDirectives);
```

Comment:
It is not obvious what is happening with annotations inside this function.
`annotations.addAll(_proceedDeprecated(fieldDirectives));` could be more clear approach